src/mongocxx/private/collection.hpp

```diff
@@ -39,6 +39,13 @@ class collection::impl {
           client_impl(client) {
     }
 
+    impl(const impl& i) : collection_t{libmongoc::collection_copy(i.collection_t)} { }
```

Comment:
I'm pretty sure bson_malloc calls abort() if malloc fails, so the program will just crash if we run out of memory.  Likewise for BSON_ASSERT.

As such I think it's safe to assume `mongoc_collection_copy` can't fail (without immediately crashing).